upload-api/test/billing.test.js

```diff
@@ -0,0 +1,96 @@
+import { createStripeBillingProvider } from '../billing.js'
+import { test } from './helpers/context.js'
+import { toEmail } from '@web3-storage/did-mailto'
+import dotenv from 'dotenv'
+
+import Stripe from 'stripe'
+import { fileURLToPath } from 'node:url'
+
+dotenv.config({ path: fileURLToPath(new URL('../../.env', import.meta.url)) })
+
+/**
+ * 
+ * @param {Stripe} stripe 
+ * @param {string} email 
+ * @returns {Promise<string | null | undefined>}
+ */
+async function getCustomerPlanByEmail(stripe, email) {
+  const customers = await stripe.customers.list({ email, expand: ['data.subscriptions'] })
+  if (customers.data.length > 1) throw new Error(`found more than one customer with email ${email}`)
+  const customer = customers.data[0]
+  if (customer) {
+    return customer.subscriptions?.data[0].items.data[0].price.lookup_key
+  }
+}
+
+/**
+ * 
+ * @param {Stripe} stripe 
+ * @param {string} email 
+ * @returns {Promise<Stripe.Customer>}
+ */
+async function setupCustomer(stripe, email) {
+  const customer = await stripe.customers.create({ email })
+
+  // set up a payment method - otherwise we won't be able to update the plan later
+  let setupIntent = await stripe.setupIntents.create({
+    customer: customer.id,
+    payment_method_types: ['card'],
+  });
+  setupIntent = await stripe.setupIntents.confirm(
+    setupIntent.id,
+    {
+      payment_method: 'pm_card_visa',
+    }
+  )
+  const paymentMethod = /** @type {string} */(setupIntent.payment_method)
+  await stripe.customers.update(customer.id, { invoice_settings: { default_payment_method: paymentMethod } })
+  return customer
+}
+
+test('stripe plan can be updated', async (t) => {
+  const stripeSecretKey = process.env.STRIPE_TEST_SECRET_KEY
```

Comment:
so this means test is dependent on a real deployment of stripe? Perhaps I would put this into the "integration" tests, so that we also run it in Post deploy? 
What do you think? I usually like to avoid to have tests that require external things, but also would like to have Stripe somehow tested

Reply:
ah, that's a good call! it'll require a bit of refactoring to make this work, so I've created a separate issue to track this refactoring: https://github.com/web3-storage/w3infra/issues/324